### ISSUE TITLE
Best Practice to log images in vision tasks

### ISSUE THREAD

**Issue by Haydnspass** [PyTorchLightning/pytorch-lightning#2953]
**What is the best practice to log images?**

Is there a standard procedure to log output images from the validation set to any kind of logger (e.g. Tensorboard)?

Usually, I like to log a number of outputs of say over the epochs to see how the prediction evolves. 
Moreover, I pick a number of random samples and log them. 

I am not quite sure how to do this with Pytorch Lightning and whether there is a common way to do it.

If not maybe I could help? My suggestion would be
- Add a Callback for logging images
- Get the indices of the samples one wants to log
- Cache these samples in `validation_step`
- Let the Callback log these images in `on_epoch_end` method

**Comment by nateraw** [comment]
Here's one way to do it...not sure if it answers your question, though.

https://github.com/PyTorchLightning/pytorch-lightning/blob/2c935d048e69a2890889dea768ecceb0252cf321/pl_examples/domain_templates/generative_adversarial_net.py#L104-L120

More specifically, these two lines of code will help ya:

```python
         grid = torchvision.utils.make_grid(sample_imgs) 
         self.logger.experiment.add_image('generated_images', grid, 0) 
```

**Comment by williamFalcon** [comment]
Here’s an example. 

https://pytorch-lightning-bolts.readthedocs.io/en/latest/vision_callbacks.html#tensorboard-image-generator

**Comment by franchesoni** [comment]
The last link provided is broken!

**Comment by zdhNarsil** [comment]
Different from @nateraw 's answer, for my case, I found `self.logger.log_image("name", [grid,])` to be useful within a LightningModule.

**Comment by SuperKenVery** [comment]
I can confirm that `self.logger.log_image("name", [grid,])` doesn't work with tensorboard

**Comment by windring** [comment]
Is there any update here?
